### PR TITLE
[LTC] Clean up unused ops

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request cleans up unused ops.

Steps:
1. tensor_aten_ops.cpp/.h.
2. WIP...

Test Plan:
lazy_tensor_core/test/cpp/build/test_ptltc

### CODE REVIEW

File: lazy_tensor_core/lazy_tensor_core/csrc/tensor_ops.cpp [hunk 1]
```diff
@@ -67,17 +67,6 @@ LazyTensor Cross(const LazyTensor& input, const LazyTensor& other,
   return lazy_tensor_aten_ops::stack({s1, s2, s3}, canonical_dim);
 }
 
-LazyTensor MakeMatrixWithDiagonal(const LazyTensor& input, int64_t diagonal) {
```
Comment (desertfire) [inline lazy_tensor_core/lazy_tensor_core/csrc/tensor_ops.cpp:70]: I see this is used by the deleted `diag` function, but I expect `diag` still needs to be supported. Is the intention of this diff to remove functions for all autogen-ed ops? If yes, maybe we should focus on ops in the `full_codegen` section.

Comment (wconstab) [reply lazy_tensor_core/lazy_tensor_core/csrc/tensor_ops.cpp:70]: I asked @alanwaketan to help clean up unused ops code, meaning code for ops that we don't yet support in the TS backend.  My rationale was that each time we do a refactor to clean up other interfaces/utils, these extra ops slow us down, and our aim is to move (most of) them to codegen anyway.  I think if we find that we need some helper function we deleted, it won't be hard to get it back from our history or pt/xla codebase; but if you prefer keeping some of it I'm fine with that too.

Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/tensor_ops.cpp:70]: All the deleted ops are not in the supported list (ts_native_functions.yaml), and not being code-generated as well.

**Review by wconstab — APPROVED** [review]

File: lazy_tensor_core/lazy_tensor_core/csrc/tensor_ops.cpp (hunk 1 above)
Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/tensor_ops.cpp:70]: Sounds ok to me. History is supposed to be tracked by git, and we can always bring things back later if needed.

**Review by desertfire — APPROVED** [review]

**Comment by alanwaketan** [comment]
Thanks for approving this pull request, @wconstab and @desertfire. Ran `test/cpp/build/test_ptltc` locally. Let's merge it.

**Pull request merged**